bfdd/bfdd_nb_config.c

```diff
@@ -483,18 +483,56 @@ int bfdd_bfd_profile_echo_mode_modify(struct nb_cb_modify_args *args)
 }
 
 /*
- * XPath: /frr-bfdd:bfdd/bfd/profile/desired-echo-echo-transmission-interval
+ * XPath: /frr-bfdd:bfdd/bfd/profile/desired-echo-transmission-interval
  */
 int bfdd_bfd_profile_desired_echo_transmission_interval_modify(
 	struct nb_cb_modify_args *args)
+{
+	struct bfd_profile *bp;
+	uint32_t min_tx;
+
+	switch (args->event) {
+	case NB_EV_VALIDATE:
+		min_tx = yang_dnode_get_uint32(args->dnode, NULL);
+		if (min_tx < 10000 || min_tx > 60000000)
+			return NB_ERR_VALIDATION;
+		break;
+
+	case NB_EV_PREPARE:
+		/* NOTHING */
+		break;
+
+	case NB_EV_APPLY:
+		min_tx = yang_dnode_get_uint32(args->dnode, NULL);
+		bp = nb_running_get_entry(args->dnode, NULL, true);
+		if (bp->min_echo_tx == min_tx)
```

Comment:
I know this is found in another places in this code, but does this check make sense? I can't recall why its there.

I think northbound won't set the same value twice so I think it might not be possible for this to happen.

Reply:
Let's keep it consistent with other code. We can check if it's possible and remove all checks in a separate PR.